airflow/models/dag.py

```diff
@@ -2435,6 +2436,51 @@ def cli(self):
         args = parser.parse_args()
         args.func(args, self)
 
+    def test(self, execution_date: datetime | None = None, run_conf=None, session=None):
+        """Execute one single DagRun for a given DAG and execution date, using the DebugExecutor."""
+
+        def add_logger_if_needed(ti: TaskInstance):
+            """
+            Add a formatted logger to the taskinstance so all logs are surfaced to the command line instead
+            of into a task file. Since this is a local test run, it is much better for the user to see logs
+            in the command line, rather than needing to search for a log file.
+            Args:
+                ti: The taskinstance that will receive a logger
+
+            """
+            format = logging.Formatter(
+                "\t[%(asctime)s] {%(filename)s:%(lineno)d} %(levelname)s - %(message)s"
+            )
+            print(f"adding logger to {ti.task_id}")
+            handler = logging.StreamHandler(sys.stdout)
+            handler.setFormatter(format)
+            if not ti.log.handlers:  # only add log handler once
+                ti.log.addHandler(handler)
+
+        execution_date = execution_date or timezone.utcnow()
+        self.clear(start_date=execution_date, end_date=execution_date, dag_run_state=False)  # type: ignore
```

Comment:
You should always pass session along to any function that accepts it

```suggestion
        self.clear(start_date=execution_date, end_date=execution_date, dag_run_state=False, session=session)
```